misc/profiles2/trekking.brf

```diff
@@ -52,7 +54,146 @@ assign is_ldcr =
+            (
+                and
+                ( not ( and reversedirection=yes cycleway:left:oneway=yes ) )
+                ( not ( and ( not reversedirection=yes ) cycleway:left:oneway=-1 ) )
+            )
+        )
+        ( or ( not cycleway:left=share_busway ) ( or ( not busway:left=opposite ) ( not busway=opposite ) ) )
+    )
+    # Along a one way path (contraflow), if we can go against traffic, there is
+    # necessarily a cycleway
+    else if badoneway then ( not isbadoneway )
+    # Any other cases where we do have a cycleway tag (two way streets),
+    # it depends on the direction of the path wrt to traffic
+    else (
+        if is_right_side_driving then (
+            or
+            ( and ( not reversedirection=yes ) ( not cycleway:left:oneway=-1 ) )
+            ( and ( reversedirection=yes ) ( cycleway:left:oneway=-1|no ) )
+        )
+        else (
+            or
+            ( and ( reversedirection=yes ) ( not cycleway:left:oneway=-1 ) )
+            ( and ( not reversedirection=yes ) ( cycleway:left:oneway=-1|no ) )
+        )
+    )
+
+assign has_right_cycleway =
+    # No cycleway hint
+    if or ( cycleway:right= ) ( cycleway:right=no ) then false
+    # Along a one way path (not contraflow), any cycleway which is not
+    # contraflow is a good hint.
+    else if goodoneway then ( and
+        (
+            and
+            ( not cycleway:right=opposite|opposite_lane|opposite_track )
+            (
+                and
+                ( not ( and reversedirection=yes cycleway:right:oneway=yes ) )
+                ( not ( and ( not reversedirection=yes ) cycleway:right:oneway=-1 ) )
+            )
+        )
+        ( or ( not cycleway:right=share_busway ) ( or ( not busway:left=opposite ) ( not busway=opposite ) ) )
+    )
+    # Along a one way path (contraflow), if we can go against traffic, there is
+    # necessarily a cycleway
+    else if badoneway then ( not isbadoneway )
+    # Any other cases where we do have a cycleway tag (two way streets),
+    # it depends on the direction of the path wrt to traffic
+    else (
+        if is_right_side_driving then (
+            or
+            ( and ( not reversedirection=yes ) ( not cycleway:right:oneway=-1 ) )
+            ( and ( reversedirection=yes ) ( cycleway:right:oneway=-1|no ) )
+        )
+        else (
+            or
+            ( and ( reversedirection=yes ) ( not cycleway:right:oneway=-1 ) )
+            ( and ( not reversedirection=yes ) ( cycleway:right:oneway=-1|no ) )
+        )
+    )
+
+# TODO: isbike could be a float value dependending on the quality of the infrastructure
+assign isbike =
+    if bicycle=yes|permissive|designated then true
+    else if lcn=yes then true
+    # Take into account cycleways
+    else if or has_cycleway ( or has_left_cycleway has_right_cycleway ) then true
+    else false
+
+# Add a penalty based on traffic speed, if there are no infrastructures
```

Comment:
I added an extra penalty for roads without cycleways based on the `maxspeed`, considering any `maxspeed` above 30 is way faster than a bike and not really safe so should be penalized.